lib/eventFactory.js

```diff
@@ -115,35 +116,59 @@ function getJobsFromPR(config) {
  * @param  {Boolean}  [config.webhooks]         If the create came from a webhook (pr or push) or not
  * @param  {Boolean}  [config.isPR]             Is it PR?
  * @param  {Object}   [config.decoratedCommit]  Decorated commit object
+ * @param  {String}   [config.rootDir]          Root directory
  * @return {Promise}
  */
 function startBuild(config) {
     /* eslint-disable global-require */
     const BuildFactory = require('./buildFactory');
     const buildFactory = BuildFactory.getInstance();
     /* eslint-enable global-require */
-    const { buildConfig, changedFiles, sourcePaths, webhooks, isPR, decoratedCommit } = config;
+    const {
+        buildConfig,
+        changedFiles,
+        sourcePaths,
+        webhooks,
+        isPR,
+        decoratedCommit,
+        rootDir
+    } = config;
     let hasChangeInSourcePaths = true;
 
     buildConfig.environment = {};
 
-    // Only check if sourcePaths exists
-    if ((webhooks || isPR) && sourcePaths) {
+    // Only check if sourcePaths or rootDir is set
+    // and webhooks or is a PR
+    if ((webhooks || isPR) && (sourcePaths || rootDir)) {
         if (!changedFiles) {
             throw new Error('Your SCM does not support Source Paths');
         }
+
+        let paths = sourcePaths || [];
+
+        // Add rootDir as a sourcePath if no sourcePaths
+        // Or prefix all sourcePaths with rootDir
+        if (rootDir) {
+            if (paths.length === 0) {
+                paths.push(`${rootDir}/`);
+            } else {
+                paths = paths.map(p => path.join(rootDir, p));
```

Comment:
I think we should not prefix `sourcePaths` with `rootDir` If user want to use `sourcePaths` along with `rootDir` let that be implicit. I can see use cases where these 2 need not be under same path.

Reply:
I'm wondering if that's confusing, since all `steps` in a `screwdriver.yaml` file would be in the context of root. I would prefer if the whole yaml was consistent.

Reply:
`steps` scoped to `rootDir` is perfectly fine. But consider a use case for a mono repo like this

```
src/[cpp/php/...]
build/[cpp/php/...]
```

making `sourcePaths` tie to `rootDir` will prevent users from using `sourcePaths` for above repo structure.

Reply:
Okay if we do it this way we need to make sure we're very explicit in the documentation about it.,